services/blockchain-indexer/shared/dataService/business/pos/stakers.js

```diff
@@ -93,36 +118,50 @@ const getStakers = async params => {
 		},
 		['stakerAddress', 'amount'],
 	);
-	if (resultSet.length) stakers.data.stakers = resultSet;
 
-	stakers.data.stakers = await BluebirdPromise.map(
-		stakers.data.stakers,
+	// Populate stakers name and prepare response
+	stakersResponse.data.stakers = await BluebirdPromise.map(
+		stakes,
 		async stake => {
-			const { name = null } = await getIndexedAccountInfo({ address: stake.stakerAddress, limit: 1 }, ['name']);
+			// Try to use cached staker name. Query DB if not found.
+			let stakerName = allowedStakersAddressNameMap[stake.stakerAddress];
+			if (!stakerName) {
+				const accountInfo = await getIndexedAccountInfo({
+					address: stake.stakerAddress,
+					limit: 1,
+				},
+				['name'],
+				);
+				stakerName = accountInfo.name;
+			}
+
 			return {
 				address: stake.stakerAddress,
 				amount: stake.amount,
-				name,
+				name: stakerName,
 			};
 		},
-		{ concurrency: stakers.data.stakers.length },
+		{ concurrency: stakes.length },
 	);
 
 	const validatorAccountInfo = await getIndexedAccountInfo(
 		{ address: params.validatorAddress, limit: 1 },
 		['name', 'publicKey'],
 	);
-	stakers.meta.validator = {
-		address: params.validatorAddress,
-		name: validatorAccountInfo.name || null,
-		publicKey: validatorAccountInfo.publicKey || null,
+	stakersResponse.meta.validator = {
+		address: stakersResponse.meta.validator.address || params.validatorAddress,
+		name: stakersResponse.meta.validator.name || validatorAccountInfo.name || null,
```

Comment:
Validator name can never be `null`.
```suggestion
		name: stakersResponse.meta.validator.name || validatorAccountInfo.name,
```

Reply:
Applied.